pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -812,3 +908,180 @@ func loadBalancerSourceRangesMatch(ic *operatorv1.IngressController, current *co
+	if subnets == nil {
+		return ""
+	}
+	joinedSubnets := ""
+	subnetCount := 0
+	for _, subnet := range subnets.IDs {
+		if subnetCount > 0 {
+			joinedSubnets += sep
+		}
+		joinedSubnets += string(subnet)
+		subnetCount++
+	}
+	for _, subnet := range subnets.Names {
+		if subnetCount > 0 {
+			joinedSubnets += sep
+		}
+		joinedSubnets += string(subnet)
+		subnetCount++
+	}
+	return joinedSubnets
+}
+
+// getAWSLoadBalancerTypeInStatus gets the AWS Load Balancer Type reported in the status.
+func getAWSLoadBalancerTypeInStatus(ic *operatorv1.IngressController) operatorv1.AWSLoadBalancerType {
+	if ic.Status.EndpointPublishingStrategy != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil {
+		return ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.Type
+	}
+	return ""
+}
+
+// getAWSClassicLoadBalancerParametersInSpec gets the ClassicLoadBalancerParameter struct
+// defined in the IngressController spec.
+func getAWSClassicLoadBalancerParametersInSpec(ic *operatorv1.IngressController) *operatorv1.AWSClassicLoadBalancerParameters {
+	if ic.Spec.EndpointPublishingStrategy != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil {
+		return ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.ClassicLoadBalancerParameters
+	}
+	return nil
+}
+
+// getAWSNetworkLoadBalancerParametersInSpec gets the NetworkLoadBalancerParameter struct
+// defined in the IngressController spec.
+func getAWSNetworkLoadBalancerParametersInSpec(ic *operatorv1.IngressController) *operatorv1.AWSNetworkLoadBalancerParameters {
+	if ic.Spec.EndpointPublishingStrategy != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil {
+		return ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.NetworkLoadBalancerParameters
+	}
+	return nil
+}
+
+// getAWSClassicLoadBalancerParametersInStatus gets the ClassicLoadBalancerParameter struct
+// reported in the IngressController status.
+func getAWSClassicLoadBalancerParametersInStatus(ic *operatorv1.IngressController) *operatorv1.AWSClassicLoadBalancerParameters {
+	if ic.Status.EndpointPublishingStrategy != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil {
+		return ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.ClassicLoadBalancerParameters
```

Comment:
I think you need to check nil for ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil this field is [optional](https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go#L542) and is a pointer .